### PR TITLE
Add code coverage for MonthCalendar

### PR DESCRIPTION
**Pull request by Zheng-Li01**
related https://github.com/dotnet/winforms/issues/10453

## Proposed changes

- Add unit tests for MonthCalendar to test its properties: Bounds, HitTest
- Add unit tests for MonthCalendar to test its methods: RemoveAllAnnuallyBoldedDates
,RemoveAnnuallyBoldedDate, RemoveAllBoldedDates, RemoveBoldedDate, RemoveAllMonthlyBoldedDates, RemoveMonthlyBoldedDate

## Customer Impact
- None

## Regression? 
-  No

## Risk
-Minimal

## Test methodology 
- Unit tests
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11263)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/MonthCalendar.CalendarNextButtonAccessibleObjectTests.cs [hunk 1]
```diff
@@ -95,4 +96,20 @@ public void CalendarNextButtonAccessibleObject_FragmentNavigate_Child_ReturnsExp
         Assert.Null(nextButton.FragmentNavigate(NavigateDirection.NavigateDirection_LastChild));
         Assert.False(control.IsHandleCreated);
     }
+
+    [WinFormsFact]
+    public void CalendarNextButtonAccessibleObject_Bounds_ReturnsExpected()
+    {
+        using MonthCalendar control = new();
```
Comment (SimonZhao888) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/MonthCalendar.CalendarNextButtonAccessibleObjectTests.cs:103]: There may be a job that you need to take care of, I found a lot of test cases are tested against the default value, please move these properties and methods test in CalendarNextButtonAccessibleObject_ctor_default for unified testing.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/MonthCalendarTests.cs [hunk 2]
```diff
@@ -4451,6 +4451,99 @@ public void CalendarCellAccessibleObject_Role_ReturnsExpected()
...
+    [WinFormsTheory]
+    [InlineData("RemoveAllAnnuallyBoldedDates", 0)]
+    [InlineData("RemoveAnnuallyBoldedDate", 1)]
+    public void MonthCalendar_RemoveAnnuallyBoldedDates_Invoke_Success(string methodName, int expectedCount)
+    {
+        using MonthCalendar calendar = InitializeCalendarWithDates();
+
+        typeof(MonthCalendar).GetMethod(methodName).Invoke(calendar, methodName == "RemoveAnnuallyBoldedDate" ? new object[] { new DateTime(2022, 1, 1) } : null);
+        calendar.AnnuallyBoldedDates.Should().HaveCount(expectedCount);
+    }
```
Comment (SimonZhao888) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/MonthCalendarTests.cs:4477]: Try to merge these 3 cases into 1.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/MonthCalendarTests.cs [hunk 3]
```diff
@@ -4451,6 +4451,99 @@ public void CalendarCellAccessibleObject_Role_ReturnsExpected()
...
+
+        typeof(MonthCalendar).GetMethod(methodName).Invoke(calendar, methodName == "RemoveMonthlyBoldedDate" ? new object[] { new DateTime(2022, 1, 1) } : null);
+        calendar.MonthlyBoldedDates.Should().HaveCount(expectedCount);
+    }
+
+    [WinFormsTheory]
+    [InlineData(0, 0)]
+    [InlineData(100, 100)]
+    [InlineData(-100, -100)]
+    public void MonthCalendar_HitTest_Int_ReturnsExpected(int x, int y)
```
Comment (SimonZhao888) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/MonthCalendarTests.cs:4505]: Pleasse merge these 3 cases into 1.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/MonthCalendar.CalendarHeaderAccessibleObjectTests.cs [hunk 4]
```diff
@@ -12,99 +13,25 @@ public class MonthCalendar_CalendarHeaderAccessibleObjectTests
...
-
         control.CreateControl();
+        control.PerformLayout();
         control.SetSelectionRange(new DateTime(2020, 8, 19), new DateTime(2020, 8, 19));
         CalendarHeaderAccessibleObject headerAccessibleObject = CreateCalendarHeaderAccessibleObject(control);
-        string actual = headerAccessibleObject.Name;
 
-        Assert.Equal("August 2020", actual);
-        Assert.True(control.IsHandleCreated);
-    }
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/MonthCalendar.CalendarHeaderAccessibleObjectTests.cs:58]: This test doesn't seem to be what is considered 'default' as it is setting selection range. Could we keep this test separate from the ctor test? I believe `headerAccessibleObject.Name` would be empty with this change.

**Review by lonitra — APPROVED** [review]
🚀

**Pull request merged**